### PR TITLE
feat: add getProcessedPendingRequests

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@umaproject.org>

**Summary**

Add more information to the return type of `getPendingRequests` (rollCount, isGovernance and lastVotingRound) and add new function `getProcessedPendingRequests` to be called statically off-chain to return the more up to date state of the pending price request

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/core/test/hardhat/data-verification-mechanism/VotingV2.js [hunk 1]
```diff
@@ -1749,9 +1748,6 @@ describe("VotingV2", function () {
     await supportedIdentifiers.methods.addSupportedIdentifier(identifier1).send({ from: accounts[0] });
     await supportedIdentifiers.methods.addSupportedIdentifier(identifier2).send({ from: accounts[0] });
 
-    // Instantiate a voting interface that supports ancillary data.
-    voting = await VotingAncillaryInterfaceTesting.at(voting.options.address);
```
Comment (md0x) [inline packages/core/test/hardhat/data-verification-mechanism/VotingV2.js:1753]: I don't think this is needed

**Review by chrismaree — APPROVED** [review]

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 2]
```diff
@@ -493,10 +493,10 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
      ****************************************/
 
     /**
-     * @notice Gets the queries that are being voted on this round.
-     * @return pendingRequests array containing identifiers of type PendingRequestAncillary.
+     * @notice Gets the requests that are being voted on this round.
+     * @return pendingRequests array containing identifiers of type PendingRequestAncillaryAugmented.
      */
-    function getPendingRequests() external view override returns (PendingRequestAncillaryAugmented[] memory) {
+    function getPendingRequests() public view override returns (PendingRequestAncillaryAugmented[] memory) {
```
Comment (mrice32) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:499]: Is this function still useful? Why not just remove view and add processResolvablePriceRequests() as the first line?

**Review by mrice32 — COMMENTED** [review]
Looks great! One minor comment

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol (hunk 2 above)
Comment (md0x) [reply packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol]: You are right @mrice32 , there was no need to have 2 functions for this. Fixed in the latest commit. Thank you!

File: packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol [hunk 3]
```diff
@@ -493,10 +493,12 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
      ****************************************/
 
     /**
-     * @notice Gets the queries that are being voted on this round.
-     * @return pendingRequests array containing identifiers of type PendingRequestAncillary.
+     * @notice Gets the requests that are being voted on this round after resolving them when possible.
+     * @return pendingRequests array containing identifiers of type PendingRequestAncillaryAugmented.
      */
-    function getPendingRequests() external view override returns (PendingRequestAncillaryAugmented[] memory) {
+    function getPendingRequests() public override returns (PendingRequestAncillaryAugmented[] memory) {
```
Comment (mrice32) [inline packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol:499]:
```suggestion
    function getPendingRequests() external override returns (PendingRequestAncillaryAugmented[] memory) {
```

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**